Source/ablastr/fields/PoissonSolver.H

```diff
@@ -57,6 +57,102 @@
 
 namespace ablastr::fields {
 
+/** Compute the L-infinity norm of the charge density `rho` across all MR levels
+ * to determine if `rho` is zero everywhere
+ *
+ * \param[in] rho The charge density a given species
+ * \param[in] finest_level The most refined mesh refinement level
+ * \param[in] absolute_tolerance The absolute convergence threshold for the MLMG solver
+ * \param[out] max_norm_rho The maximum L-infinity norm of `rho` across all levels
+ */
+inline amrex::Real getMaxNormRho (
+    amrex::Vector<amrex::MultiFab*> const & rho,
+    int finest_level,
+    amrex::Real & absolute_tolerance)
+{
+    amrex::Real max_norm_rho = 0.0;
+    for (int lev=0; lev<=finest_level; lev++) {
+        max_norm_rho = amrex::max(max_norm_rho, rho[lev]->norm0());
+    }
+    amrex::ParallelDescriptor::ReduceRealMax(max_norm_rho);
+
+    const bool always_use_bnorm = (max_norm_rho > 0);
+    if (!always_use_bnorm) {
```

Comment:
```suggestion
    if (max_norm_rho == 0) {
```

Reply:
I considered this but decided against it since equality comparisons between floats are not great. But his check is really meant to catch the case that we have a true vacuum (not perfect neutrality), so the equality check is probably okay.

Reply:
I think that `== 0.` is fine (but only for zero, not for other float values). We use this in other places in WarpX.